app/markdownhighlighter.cpp

```diff
@@ -99,89 +105,110 @@ void MarkdownHighlighter::highlightBlock(const QString &textBlock)
         return;
     }
 
-    workerThread->enqueue(text);
+    // cut YAML headers
+    QString actualText;
+    unsigned long offset = 0;
+    if (yamlHeaderSupportEnabled) {
+        YamlHeaderChecker checker(text);
+        actualText = checker.body();
+        offset = checker.header().length();
```

Comment:
I would see it as a task of the YamlHeaderChecker class to return the offset. Something like `checker.bodyOffset()`.